### PR TITLE
[FIX] Load images once in parallel function instead of single function

### LINKED ISSUE #5150: [ENH] Check whether `load_img` loads images in a memory mapped way
I think we should deal with this separately, but I wanted to test our hypothesis that joblib + memory mapped in_memory image is what's causing the path vs. in_memory difference. 

So I compared the image loaded via `load_img` (I think it doesn't use dataobj) vs. `concat_imgs` (defo uses dataobj):

With `load_img` the difference between NiftiMasker in_memory, numpy masker path and numpy masker in_memory disappears: https://output.circle-artifacts.com/output/job/c99698a5-290c-4ee7-9750-1e551c4774fa/artifacts/0/dev/auto_examples/07_advanced/plot_mask_large_fmri.html#conclusion

_Originally posted by @man-shu in https://github.com/nilearn/nilearn/issues/5113#issuecomment-2654402188_

### PR DESCRIPTION
**Pull request by man-shu**
- Closes #5150

Changes proposed in this pull request:

- load the image once in "parallel" functions
- "numpy path" variant is not relavant anymore
- proxy images/memory mapping isn't relavant because they have the same peak memory usage as array images:
  - probably because full array is needed for masking
  - lower memory usage should only be expected for use cases where segments of the array are needed

### CODE REVIEW

**Review by bthirion — APPROVED** [review]
LGTM, thx.

File: examples/07_advanced/plot_mask_large_fmri.py [hunk 1]
```diff
@@ -116,61 +110,73 @@
...
+def nifti_masker_single(fmri, mask):
+    return NiftiMasker(mask_img=mask).fit_transform(fmri)
+
+
+def nifti_masker_parallel_path(fmri_path, mask_paths):
+    Parallel(n_jobs=N_REGIONS)(
+        delayed(nifti_masker_single)(fmri_path, mask_path)
+        for mask_path in mask_paths
+    )
+    return 1
```
Comment (emdupre) [inline examples/07_advanced/plot_mask_large_fmri.py:153]: Just to make sure I understand : Why are we adding these `return` statements ?

File: examples/07_advanced/plot_mask_large_fmri.py [hunk 2]
```diff
@@ -320,27 +289,27 @@ def numpy_masker_shared_parallel(img, masks):
     [
         "NiftiMasker,\nwith path",
         "NiftiMasker,\nwith in-memory\nimage",
-        "Numpy indexing,\nwith path",
-        "Numpy indexing,\nwith in-memory\nimage",
+        "Numpy indexing",
         "Numpy indexing,\nwith shared\nmemory",
```
Comment (emdupre) [inline examples/07_advanced/plot_mask_large_fmri.py:293]: Light suggestion to be clearer that we're referring to the class `SharedMemory` rather than the idea of shared memory
```suggestion
        "Numpy indexing,\nwith\nSharedMemory",
```

**Review by emdupre — APPROVED** [review]
Thanks, @man-shu ! This pattern of results makes a bit more sense to me now. I left two light comments / clarifications, but I don't think we need a full re-review for those.

File: examples/07_advanced/plot_mask_large_fmri.py (hunk 1 above)
Comment (man-shu) [reply examples/07_advanced/plot_mask_large_fmri.py:153]: Nothing specific, I just wanted to avoid returning the output of Parallel and later storing it in a variable because that would show up as an irrelavant memory peak.
